core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -366,9 +366,12 @@ private[spark] class MesosClusterScheduler(
     } else {
       ""
     }
+
+    val sslOps = conf.getSSLConf.toArray.map({ case (k, v) => s"-D$k=$v" }).mkString(" ")
     val envBuilder = Environment.newBuilder()
     desc.command.environment.foreach { case (k, v) =>
-      envBuilder.addVariables(Variable.newBuilder().setName(k).setValue(v).build())
+      val vv = if (k == "SPARK_SUBMIT_OPTS") s"$v $sslOps" else v
```

Comment:
SPARK_SUBMIT_OPTS is undocumented, and I don't trust it.  Let's use the documented method of passing conf variables to spark-submit: `--conf`.

Also, this will only result in the ssl variables being set if the user has already set the SPARK_SUBMIT_OPTS env variable, which is incorrect.  We want them set in all cases.